connaisseur/alert.py

```diff
@@ -48,6 +48,166 @@ def alerting_required(self, event_category: str) -> bool:
         return bool(self.config.get(event_category))
 
 
+class AlertReceiverAuthentication:
+    """
+    Class to store authentication information for securely sending events to the alert receiver.
+    """
+
+    class AlertReceiverBasicAuthentication:
+        """
+        Class to store authentication information for basic authentication type with username and password.
+        """
+        username: str
+        password: str
+        authentication_type: str
+        authorization_prefix: str = "Basic"
+
+        def __init__(self, alert_receiver_config: dict):
+            basic_authentication_config = alert_receiver_config.get(
+                "receiver_authentication_basic", None
+            )
+
+            if (
+                basic_authentication_config is None
+            ):  # TODO maybe remove this check since it is included in the json validation?
+                raise ConfigurationError("No basic authentication configuration found.")
+
+            username_env = basic_authentication_config.get("username_env", None)
+            password_env = basic_authentication_config.get("password_env", None)
+
+            if (
+                username_env is None or password_env is None
+            ):  # TODO maybe remove this check since it is included in the json validation?
+                raise ConfigurationError(
+                    "No username_env or password_env configuration found."
+                )
+
+            self.username = os.environ.get(username_env, None)
+            self.password = os.environ.get(password_env, None)
+
+            if self.username is None or self.password is None:
+                raise ConfigurationError(
+                    f"No username or password found from environmental variables {username_env} and {password_env}."
+                )
+
+            self.authorization_prefix = basic_authentication_config.get(
+                "authorization_prefix", "Basic"
+            )
+            # TODO maybe validate authorization prefix
+
+        def get_header(self) -> dict:
+            return {
+                "Authorization": f"{self.authorization_prefix} {self.username}:{self.password}"
+            }
+
+    class AlertReceiverBearerAuthentication:
+        """
+        Class to store authentication information for bearer authentication type which uses a token.
+        """
+        token: str
+        authorization_prefix: str = "Bearer"  # default is bearer
+
+        def __init__(self, alert_receiver_config: dict):
+            bearer_authentication_config = alert_receiver_config.get(
+                "receiver_authentication_bearer", None
+            )
+
+            if (
+                bearer_authentication_config is None
+            ):  # TODO maybe remove this check since it is included in the json validation?
+                raise ConfigurationError(
+                    "No bearer authentication configuration found."
+                )
```

Comment:
This is boilerplate (together with assigning `self.authentication_prefix` at the end of the method).

Suggestion: make a base class, introduce a variable for the lookup key (e.g. `"receiver_authentication_bearer"`), and run all this via `super().__init__(*args, **kwargs)`.